### PR TITLE
Remove custom __getattr__ and __setattr__ from gpytorch.Module (addresses #180)

### PR DESCRIPTION
**Pull request by jacobrgardner**
It seems like this was easy enough -- I think these really only existed to stop people from creating parameters without bounds. Since bounds are now handled exclusively as priors, none of the functionality in those methods accomplished anything necessary.

Travis, what do you think?

#180

### CODE REVIEW

File: test/examples/test_kissgp_gp_classification.py [hunk 1]
```diff
@@ -28,11 +28,7 @@ def __init__(self):
         self.covar_module = RBFKernel(
             log_lengthscale_prior=SmoothedBoxPrior(exp(-5), exp(6), sigma=0.1, log_transform=True)
         )
-        self.register_parameter(
-            name="log_outputscale",
-            parameter=torch.nn.Parameter(torch.Tensor([0])),
-            prior=SmoothedBoxPrior(exp(-5), exp(6), sigma=0.1, log_transform=True),
```
Comment (Balandat) [inline test/examples/test_kissgp_gp_classification.py]: Did you mean to get rid of the prior here?

Comment (jacobrgardner) [reply test/examples/test_kissgp_gp_classification.py]: Can we keep the prior and still use `self.my_param = ...` to create a parameter?

Comment (Balandat) [reply test/examples/test_kissgp_gp_classification.py]: Well you can, but you'd have to manually call `set_parameter_priors` to set the prior. The `self.my_param = ...` syntax doesn't really allow for a clean way to set a prior I think.

Comment (jacobrgardner) [reply test/examples/test_kissgp_gp_classification.py]: Yeah, I changed it here so that we had one unittest that did things that way

**Pull request merged**